ui/src/app/metadata/detail/metadata-list/metadata-list.component.ts

```diff
@@ -164,9 +165,32 @@ export class MetadataListComponent extends AbstractListComponent<IJobCode | INam
     }
   }
 
-  private handleClickDeleteItem(metadata: IJobCode | INamedReference | undefined): void {
+  private handleDeleteMultipleNamedReferences(namedReferences: NamedReferenceInterface[], index: number, notDeleted = 0): void {
+    if (index < namedReferences.length) {
+      this.namedReferenceService.deleteNamedReferenceWithResult(namedReferences[index].id ?? 0).subscribe(data => {
+        if (data && data.success) {
+          this.handleDeleteMultipleNamedReferences(namedReferences, index + 1, notDeleted)
+        } else if (data && !data.success) {
+          this.handleDeleteMultipleNamedReferences(namedReferences, index + 1, notDeleted + 1)
+        }
+      })
+    } else {
```

Comment:
This code is very similar to job codes.
We can have a new function like this

```
  private onDeleteMultipleFinish(notDeleted: number, metadataDeleted: string): void {
    this.toastService.hideBlockingLoader()
    if (notDeleted > 0) {
      this.toastService.showToast("Warning", `Some ${metadataDeleted} could not be deleted`)
    } else {
      this.toastService.showToast("Success", `All selected ${metadataDeleted} have been deleted`)
    }
    this.loadNextPage()
  }
```

and use it in JobCodes

```
else {
      this.onDeleteMultipleFinish(notDeleted, "Occupations")
    }
```

and in Named References 

```
else {
      this.onDeleteMultipleFinish(notDeleted, "Named References")
    }
```

What do you think?

Reply:
Yeah sounds good, but lets not hardcode the second arg, we should use MetadataSelected instead.